sklearn/base.py

```diff
@@ -889,6 +890,18 @@ def fit_transform(self, X, y=None, **fit_params):
                     UserWarning,
                 )
 
+        for key, _ in fit_params.items():
+            if (
+                key not in inspect.signature(self.fit).parameters
+                and "fit_params" not in inspect.signature(self.fit).parameters
+            ):
+                message = (
+                    f"`{self.__class__.__name__}.fit` got an unexpected keyword "
+                    f"argument `{key}`. Only pass {key} into `fit_transform` if `fit` "
+                    "expects them or can handle `**fit_params`."
+                )
+                raise TypeError(message)
```

Comment:
I think instead of doing this manually, you can rely on python's machinery, and put a try/except around call to `fit` and catch `TypeError`, and then do a `raise ... from e` with a `TypeError` and a nicer error message.